packages/ui/src/primitives/tailwind/Select/index.tsx

```diff
@@ -36,7 +36,12 @@ import Input from '../Input'
 
 export type OptionValueType = string | number
 
-export type SelectOptionsType = { label: string; value: any; disabled?: boolean }
+export type SelectOptionsType = {
+  label: string
+  value: any
+  disabled?: boolean
+  metadata?: { [key: string]: string }
+}
```

Comment:
Going forward, we want to use `MenuItem` pattern to render Options. This will allow us to exert greater control over how the options are rendered. Something like this: https://github.com/ir-engine/ir-engine/blob/dev/packages/ui/src/primitives/tailwind/Dropdown/index.tsx.

It can be used as:
```jsx
<Select>
  <MenuItem value={...}>labelContent</MenuItem>
  <MenuItem value={...}>labelContent</MenuItem>
  <MenuItem value={...}>labelContent</MenuItem>
</Select>
```
This is inspired from: https://mui.com/material-ui/react-select/

Reply:
Right, I 100% agree that such is the best approach, by doing that it would avoid the multiple iterations currently existing within the select component, the thing is that this is a 0.9 requirement, I did spent quite a bit of time analyzing the options, in the short and the long term.

I concluded that by making completely disrupting changes would add a lot of risk for the time being, so I decided to implement something that wouldn't break all of the other current implementations of the Select component across all apps while achieving the immediate requirement.

If you analyze my approach, basically I'm by passing the current implementation with customMenu in a way that by implementing it is basically doing what you are mentioning but without breaking other Select implementations that still are using the select component's internal rendering.

I guess the question here is: Is the <Select><MenuItem></MenuItem></Select> approach a 0.9 requirement? depending on the answer comes the decision.

IF the answer is yes then we can simply migrate what's inside the customMenu.
IF the answer is no then we can use this as a temporary solution while the <Select><MenuItem></MenuItem></Select> is available.

In any case, my implementation has always been consider under my view as a low risk immediate solution for 0.9, having a subcomponent like MenuItem it definitely has been my long term solution and I wanted to revisit after 0.9.

Reply:
`release0.9` is released yesterday and PRs are no longer made to that branch. Nevertheless all UI refactoring PRs are targeted to dev branch (not release branch).

Reiterating again: to fix the Phone number Input quickly, I think a separate component such as `<PhoneInput />` can be designed and it can be inside multitenancy repo for now. This way, `<Select>` will be left untouched and hence not breaking any other implementations.